x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/components/utils.test.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
```

Comment:
I have a few issues with relying on moment in test files for both the input and output - it feels like circular/non-"unit" testing, and in this case I have no idea what `moment().format('ll')` looks like without going to moment's site/documentation.

The latter gets somewhat resolved by switching to `<FormattedRelative />` and `<FormattedDateTime />` (FormattedDateTime has its own unit tests, so we don't need to dive into it & can see an example output in its tests), so we can now can simply check for which component was rendered.

For the `isDateToday` portion of the logic, since we're no longer using moment, the way we should go about testing & documenting that piece of logic is via statically mocking the `Date.now()` returned value. Here's a few other example test files across App Search where we mock the global Date:

- [api_logs_logic](https://github.com/elastic/kibana/blob/12986fb8fc46a7adde34ae3052104b29c6f74c69/x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/api_logs_logic.test.ts#L146-L150)
- [api_logs/utils](https://github.com/elastic/kibana/blob/56d9f3d96832f67283ee4a35e306f89fa59ad2f4/x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/utils.test.ts#L13-L15)
- [enterprise_search_request_handler](https://github.com/elastic/kibana/blob/58b1416f845cdcdfa615533d8c09bbc30a4624a1/x-pack/plugins/enterprise_search/server/lib/enterprise_search_request_handler.test.ts#L434-L438)

In this scenario, I'd suggest something like this:

```js
import { FormattedRelative } from '@kbn/i18n/react';

import { FormattedDateTime } from '../../../utils/formatted_date_time';

import { determineTimestampDisplay } from './utils';

describe('determineTimestampDisplay', () => {
  const mockToday = jest
    .spyOn(global.Date, 'now')
    .mockImplementation(() => new Date('1970-01-02').valueOf());

  afterAll(() => mockToday.mockRestore());

  it('uses a relative time format (x minutes ago) if the timestamp is from today', () => {
    const timestamp = determineTimestampDisplay('1970-01-02');

    expect(timestamp.type).toEqual(FormattedRelative);
    expect(timestamp.props.value).toEqual(new Date('1970-01-02'));
  });

  it('uses a date if the timestamp is before today', () => {
    const timestamp = determineTimestampDisplay('1970-01-01');

    expect(timestamp.type).toEqual(FormattedDateTime);
    expect(timestamp.props.date).toEqual(new Date('1970-01-01'));
  });
});
```

As an aside, I tend to use Jan 1970 (the beginning of the all time :trollface:) instead of current dates or years as it's easier to treat it more clearly as an example/mock, and also it dates code less obviously when you come back to it in a few years.

Reply:
This has been addressed in eac071f